pymc3/distributions/multivariate.py

```diff
@@ -1646,16 +1650,21 @@ def dist(
         rowchol=None,
         colcov=None,
         colchol=None,
-        shape=None,
         *args,
         **kwargs,
     ):
 
         cholesky = Cholesky(lower=True, on_error="raise")
 
-        if mu.ndim == 1:
-            raise ValueError(
-                "1x1 Matrix was provided. Please use Normal distribution for such cases."
+        if "size" in kwargs:
+            raise NotImplementedError("MatrixNormal doesn't support size argument")
```

Comment:
Turns out `size` was never supported in `MatrixNormal` in `v3`. (Thanks to @ricardoV94 for pointing that out.

Reply:
Users may also pass shape (and invalid shapes) or "dims", so this check might not be enough. Can a ndim check be done inside the logp method instead? And what about the RV, does it work properly for non-default sizes?

Reply:
Also just to be clear 2d shapes (n x m) were allowed, just not "replication" shapes. I am not sure if that maps to any non-`None` size because I don't really grasp what size means for multivariate distributions.

Reply:
I added a warning for `shape` since if only `(m x n)` shape is allowed we can directly derive `m` and `n` from parameters. 

> And what about the RV, does it work properly for non-default sizes?

What do you mean by non-default sizes?

Reply:
If you used the RV directly with some size like (2, 3, 3), would it do a sensible thing in the `rng_fn`? Or should it also raise there?

Reply:
The RV can handle sizes like that because the main operation over there i.e. `np.matmul` will treat them as stack of matrices (which is what we intend) so I don't think we need to raise there.

Reply:
I think size is one of the default arguments for distributions. Perhaps it's better to simply check it is `None`